core-client/transports/src/transports/duplex.rs

```diff
@@ -214,12 +214,27 @@ where
 									);
 								}
 							} else {
-								log::warn!(
-									"The response does not have expected subscription id in response: {:?} ({:?}): {:?}",
+								let err = RpcError::Other(format_err!(
+									"Subscription {:?} ({:?}) rejected: {:?}",
 									id,
 									method,
 									result,
-								);
+								));
+								match subscription.channel.poll_ready() {
+									Ok(Async::Ready(())) => {
+										subscription
+											.channel
+											.try_send(result)
+											.expect("The channel is ready; qed");
+									}
+									Ok(Async::NotReady) => {
+										self.incoming.push_back((id, result, Some(method), sid));
```

Comment:
What if it's the only request that is left? What are the chances that we will just loop forever on that item?

Reply:
It would loop as long as `subscription.channel` is `NotReady`, so as soon as the response channel is not full it would be added.

Reply:
Wouldn't that end up in busy-waiting for that channel to be ready? Also if we have single-threaded runtime it probably won't ever happen, since we are busy handling this `poll`, so we can't really process any of the responses.

Reply:
Wait I forgot about the `break` down there V V V